src/lexer/regexFactory.ts

```diff
@@ -17,11 +11,13 @@ export const lineComment = (lineCommentTypes: string[]) =>
   new RegExp(`(?:${lineCommentTypes.map(escapeRegExp).join('|')}).*?(?=\r\n|\r|\n|$)`, 'uy');
 
 /**
- * Builds a RegExp for matching parenthesis patterns, escaping them with `escapeParen`
- * @param {string[]} parens - list of strings that denote parenthesis patterns
+ * Builds a RegExp for matching either open- or close-parenthesis patterns
  */
-export const parenthesis = (parens: string[]): RegExp =>
-  patternToRegex(parens.map(escapeParen).join('|'));
+export const parenthesis = (kind: 'open' | 'close', extraParens: ('[]' | '{}')[] = []): RegExp => {
```

Comment:
can we move this `'[]' | '{}'` to a ParenType `type` in regexTypes ?

Reply:
Opted to skip this for now. It's just two places where this type is used and it's a very simple one.